lib/si-data-pg/src/lib.rs

```diff
@@ -2888,3 +2915,40 @@ impl PgOwnedTransaction {
 async fn test_connection_infallible_and_warm_up_pool_task(check_pool: PgPool) {
     let _result = check_pool.test_connection().await;
 }
+
+fn evict_old_and_idle_pgpool_connections(
+    pg_pool: Pool,
+    check_interval: Duration,
+    max_age: Duration,
+    max_idle_time: Duration,
+) -> impl Future<Output = ()> {
```

Comment:
Yeah, this'll work, although you may also be able to make this function signature `async fn evict_old_and_idle_pgpool_connections(...) {}`, meaning you can drop the `async move {}` surrounding scope inside the function. My guess would be this might have been an editor refactoring helper? I might be missing something though!

Reply:
I didn't know you could just pass a promise to tokio spawn, so I copied the block! Thanks for the callout

Reply:
Pretty wicked, huh?